src/main/java/com/spotify/folsom/ketama/SrvKetamaClient.java

```diff
@@ -99,17 +98,33 @@ public void updateDNS() {
       if (shutdown) {
         return;
       }
-      List<HostAndPort> newAddresses = Ordering.from(HostAndPortComparator.INSTANCE)
-              .sortedCopy(srvResolver.resolve(srvRecord));
-      if (!newAddresses.equals(addresses)) {
-        addresses = newAddresses;
-        log.info("Connecting to " + newAddresses);
-        List<AddressAndClient> addressAndClients = getAddressesAndClients(newAddresses);
-        setPendingClient(addressAndClients);
+      long ttl = Integer.MAX_VALUE;
+      try {
+        List<LookupResult> lookupResults = srvResolver.resolve(srvRecord);
+        List<HostAndPort> hosts = Lists.newArrayListWithCapacity(lookupResults.size());
+        for (LookupResult lookupResult : lookupResults) {
+          hosts.add(HostAndPort.fromParts(lookupResult.host(), lookupResult.port()));
+          ttl = Math.min(ttl, lookupResult.ttl());
+        }
+        List<HostAndPort> newAddresses = Ordering.from(HostAndPortComparator.INSTANCE)
+                .sortedCopy(hosts);
+        if (!newAddresses.equals(addresses)) {
+          addresses = newAddresses;
+          log.info("Connecting to " + newAddresses);
+          List<AddressAndClient> addressAndClients = getAddressesAndClients(newAddresses);
+          setPendingClient(addressAndClients);
+        }
+      } finally {
+        long delay = clamp(MIN_DNS_WAIT_TIME, MAX_DNS_WAIT_TIME, ttl);
+        refreshJob = this.executor.schedule(refreshRunnable, delay, TimeUnit.SECONDS);
```

Comment:
Seems we'd wait 1 hour before attempting to resolve again if there was no records found. That'll happen from time to time in prod due to infra breakage.